lib/puppet/type/truststore_certificate.rb

```diff
@@ -0,0 +1,47 @@
+Puppet::Type.newtype(:truststore_certificate) do
+  desc 'adds a certificate to a pkcs12 truststore'
+
+  ensurable
+
+  def self.title_patterns
+    [ [ /(.+):(.+)/m, [ [:truststore], [:alias] ] ] ]
+  end
+
+  newparam(:alias, :namevar => true) do
+    desc "The certificate alias used to store inside the truststore"
+  end
+
+  newparam(:truststore, :namevar => true) do
+    desc "Path to the truststore to use or create when importing the certificate"
+    isrequired
+  end
+
+  newproperty(:certificate) do
+    desc "Path to the certificate to add to the truststore"
+
+    def fingerprint(file)
+      provider.fingerprint(file)
+    end
+
+    def should_to_s(newvalue)
+      self.class.format_value_for_display(fingerprint(newvalue))
+    end
+
+    def insync?(is)
+      is == fingerprint(should)
+    end
+  end
+
+  newparam(:password_file) do
+    desc "Path to file containing the truststore password"
+    isrequired
+  end
+
+  autorequire(:file) do
+    [self[:password_file], File.dirname(self[:truststore]), self[:certificate]]
+  end
+
+  autonotify(:file) do
```

Comment:
Linking back: it was impossible to write a test for this using rspec-puppet. https://github.com/rodjek/rspec-puppet/commit/7484feb56ffa1380ffe778c7f9f89daa06c07570 has been merged but is unreleased. Once it's merged, it should be possible to test this:
```ruby
it { is_expected.to contain_truststore_certificate('the-cert').that_notifies('File[/truststore]') }
```

Reply:
Maybe file an issue to remember this? Or just open a draft PR with this added?